components/data-provider/src/structs/dates.rs

```diff
@@ -0,0 +1,131 @@
+        _ => None,
+    }
+}
+
+pub mod gregorian {
+    use serde::{Deserialize, Serialize};
+    use std::borrow::Cow;
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct DatesV1 {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub months: Option<months::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub days: Option<days::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub day_periods: Option<day_periods::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub date_style_patterns: Option<patterns::StylePatternsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub time_style_patterns: Option<patterns::StylePatternsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub date_time_style_patterns: Option<patterns::StylePatternsV1>,
+
+        #[serde(skip_serializing_if = "Vec::is_empty")]
+        pub available_formats: Vec<(Cow<'static, str>, Cow<'static, str>)>,
+    }
+
+    macro_rules! display_names {
+        ($name: ident, $($element: ident: $ty: ty),*) => {
+            pub mod $name {
+                use super::*;
+
+                #[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+                pub struct DisplayNamesV1 {
+                    $(pub $element: $ty),*
+                }
+
+                #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+                pub struct WidthsV1 {
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub abbreviated: Option<DisplayNamesV1>,
+
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub narrow: Option<DisplayNamesV1>,
+
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub short: Option<DisplayNamesV1>,
+
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub wide: Option<DisplayNamesV1>,
+                }
+
+                #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+                pub struct ContextsV1 {
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub format: Option<WidthsV1>,
+
+                    #[serde(skip_serializing_if = "Option::is_none")]
+                    pub stand_alone: Option<WidthsV1>,
+                }
+            }
+        }
+    }
+
+    display_names!(
+        months,
+        m1: Cow<'static, str>,
+        m2: Cow<'static, str>,
+        m3: Cow<'static, str>,
+        m4: Cow<'static, str>,
+        m5: Cow<'static, str>,
+        m6: Cow<'static, str>,
+        m7: Cow<'static, str>,
+        m8: Cow<'static, str>,
+        m9: Cow<'static, str>,
+        m10: Cow<'static, str>,
+        m11: Cow<'static, str>,
+        m12: Cow<'static, str>
```

Comment:
I put names as they are in CLDR, but since this is gregorian calendar, I'm not sure if this is the right thing to do. @mihnita @sffc ?

Reply:
Would you recommend the same for weekdays? `[Cow<'static, str>; 7]`?

Reply:
Yes

Reply:
Done